test/jdk/javax/swing/JTable/JTableScrollPrintTest.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Component;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.print.PageFormat;
+import java.awt.print.Paper;
+import java.awt.print.Printable;
+import java.awt.print.PrinterException;
+import java.awt.print.PrinterJob;
+
+import javax.swing.BoxLayout;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTable;
+import javax.swing.JViewport;
+import javax.swing.SwingUtilities;
+import javax.swing.table.DefaultTableModel;
+
+/*
+ * @test
+ * @key headful
+ * @bug 8210807
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @summary Test to check if JTable can be printed when it is
+ * added JScrollPane.
+ * @run main/manual JTableScrollPrintTest
+ */
+
+public class JTableScrollPrintTest {
+    public static JFrame frame;
+    public static PassFailJFrame passFailJFrame;
+
+    public static void main(String[] args) throws Exception {
+        SwingUtilities.invokeAndWait(() -> {
+            try {
+                initialize();
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        });
+        passFailJFrame.awaitAndCheck();
+    }
+
+    public static void initialize() throws Exception{
+        final String INSTRUCTIONS = """
+                Instructions to Test:
+                1. Print table onto Paper/PDF, using the Print Dialog.
+                2. If entire table is printed, then the Test is PASS.
+                3. If table is partially printed without table cells,
+                then the Test is FAIL.
+                """;
+        TestTable testTable = new TestTable(true);
+        frame = new JFrame("JTable Print Test");
+        passFailJFrame = new PassFailJFrame("Test Instructions", INSTRUCTIONS, 5L, 6, 35);
```

Comment:
If you want then you can use builder pattern for configuring PassFailJFrame.

Reply:
Not required I guess.

Reply:
Not required really, just a suggestion.